src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.Tests/ServiceProviderEngineScopeTests.cs

```diff
@@ -9,13 +11,28 @@ namespace Microsoft.Extensions.DependencyInjection.ServiceLookup
     public class ServiceProviderEngineScopeTests
     {
         [Fact]
-        public void Dispose_DoesntClearResolvedServices()
+        public void ResolvedServicesAfterDispose_ThrowsObjectDispose()
```

Comment:
This is the biggest change. Accessing services after disposing will throw.

Reply:
I guess this [still is a regression test](https://github.com/dotnet/runtime/pull/50463/files#r605731182).